### DISCUSSION TITLE
Enquiries Form and RLS

### DISCUSSION THREAD

**Discussion by nbx3** [supabase/supabase#4738] [Questions] [answered]
Just trying out supabase, its great, but in my try out project, i've run into a problem. 

I'm building a simple backend for a form submission. The form needs to be submitted without user authentication. I've created two tables, form_data and form_data_store - the form js / api pushes data into the form_data table which has no security on it and i use the anon key to do this. 

Because i dont want the bad guys to steal my data, i decided i should copy the data using a PG function and trigger into the form_data_store table and then delete the original data in form_data table. I've been able to setup these correctly and test this in the supabase gui - i do a new insert on the form_data table and it then appears in the form_data_store table. 

However, If i do this using the API then i get an error "new row violates row-level security policy for table \"form_data_store\""

Is there a way to resolve this? Is it a bug or the correct behaviour from supabase pgsql? I would also like to know if there is a better way of solving this problem. Thanks

**Comment by GaryAustin1** [comment, marked as answer]
If you are comfortable with anyone who has access to the function using it then you add "security definer" to the function so it runs with the permissions of the owner of the function (no RLS in most cases) instead of the user.
https://supabase.com/docs/guides/auth/row-level-security#policies-with-security-definer-functions

**Reply by nbx3** [reply]
Thanks that resolved it perfectly.
